packages/lib-classifier/src/store/SubjectViewerStore/SubjectViewerStore.js

```diff
@@ -47,43 +46,28 @@ const SubjectViewer = types
       }
       return false
     },
-    
+ 
+    get hasAnnotateTask () {
+      const canAnnotate = getRoot(self)?.workflowSteps.hasAnnotateTask
+
+      // Make sure the right button is active in the ImageToolbar
+      if (canAnnotate) {
+        self.enableAnnotate()
+      } else {
+        self.enableMove()
+      }
+
+      return canAnnotate
+    },
+
     get interactionMode () {
       // Default interaction mode is 'annotate'
       return (!self.annotate && self.move) ? 'move' : 'annotate'
     }
   }))
 
   .actions(self => {
-    function createSubjectObserver () {
-      const subjectDisposer = autorun(() => {
-        const validSubjectReference = isValidReference(() => getRoot(self).subjects.active)
-        if (validSubjectReference) {
-          const subject = getRoot(self).subjects.active
-          self.resetSubject(subject)
-        }
-      }, { name: 'SubjectViewerStore Subject Observer autorun' })
-      addDisposer(self, subjectDisposer)
-    }
```

Comment:
I'm pretty sure you need this reaction. It resets the viewer when the active subject changes.

Reply:
Resetting the subject like this might also mean that this model is managing per-subject state that is better off stored in individual subjects. However, rewriting the data model is outside the scope of this PR. 

Whenever you see `getRoot(self)` in a model, that’s a code smell implying that the data model isn’t quite right. 

Nodes in the tree should prefer to inherit state from their parents, and rarely interact with siblings.

Reply:
I've re-added this reaction.